packages/data-service-generator/src/server/secrets-manager/create-secrets-manager.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  CreateServerSecretsManagerParams,
+  EventNames,
+  ModuleMap,
+  SecretsNameKey,
+} from "@amplication/code-gen-types";
+import { print } from "@amplication/code-gen-utils";
+import pluginWrapper from "../../plugin-wrapper";
+import DsgContext from "../../dsg-context";
+import { builders, namedTypes } from "ast-types";
+import { createDTOFile } from "../resource/dto/create-dto-module";
+import { pascalCase } from "pascal-case";
+import path from "path";
+import fg from "fast-glob";
+import { promises as fsPromises } from "fs";
+import { getFileEncoding } from "../../utils/get-file-encoding";
+
+export function createSecretsManager(
+  eventParams: CreateServerSecretsManagerParams
+): ModuleMap {
+  return pluginWrapper(
+    createSecretsManagerInternal,
+    EventNames.CreateServerSecretsManager,
+    eventParams
+  );
+}
+
+export async function createSecretsManagerInternal({
+  secretsNameKey,
+}: CreateServerSecretsManagerParams): Promise<ModuleMap> {
+  const context = DsgContext.getInstance;
+
+  const { serverDirectories } = DsgContext.getInstance;
```

Comment:
It seems not "clean" to call getInstance twice instead of using the `const context` for the `{ serverDirectories }` as well

Reply:
Fixed. Good spot!